generator.py

```diff
@@ -11,32 +10,29 @@
 NEED_UPDATE = 1
 FAILURE = 2
 
-DEFAULT_GIR_FILES_DIRECTORY = 'gir-files'
-DEFAULT_GIR_PATH = './gir/target/release/gir'
+DEFAULT_GIR_FILES_DIRECTORY = Path('./gir-files')
+DEFAULT_GIR_DIRECTORY = Path('./gir/')
+DEFAULT_GIR_PATH = DEFAULT_GIR_DIRECTORY / 'target/release/gir'
 
 
 def run_command(command, folder=None):
     if folder is None:
         folder = "."
-    child = subprocess.Popen(command, cwd=folder)
-    child.communicate()
-    if child.returncode != 0:
-        print("Command `{}` failed with return code `{}`...".format(command, child.returncode))
+    ret = subprocess.run(command, cwd=folder)
+    if ret.returncode != 0:
+        print("Command `{}` failed with `{}`...".format(command, ret))
         return False
     return True
 
 
 def update_workspace():
-    try:
-        return run_command(['cargo', 'build', '--release'], 'gir')
-    except:
-        return False
+    return run_command(['cargo', 'build', '--release'], 'gir')
 
 
 def ask_yes_no_question(question, conf):
     question = '{} [y/N] '.format(question)
     if conf.yes:
-        print(question)
+        print(question + 'y')
         return True
     if sys.version_info[0] < 3:
         line = raw_input(question)
```

Comment:
Oops, there is no function `raw_input()` anywhere. Glad we're using Python so nobody noticed this ;)

Reply:
@sdroege Should be in Python 2, which is what this code falls back to?

Then again Python 2 is deprecated for almost a year now, and the shebang specifies Python 3, might as well remove this in a followup PR :)

Reply:
Oh I just had my editor complain that this function doesn't exist (and I don't have Python 2 installed)

Reply:
The code uses `Path` & some new fancy python 3 features, it should be safe to remove the python 2 fallback I would say.

Reply:
wouuuups

Reply:
Not a problem really, but we should remove the Python 2 backwards compat code. This is 2020.

Reply:
Absolutely, it's just that I'm so used to write this code that I didn't pay attention. Sending a PR.